### ISSUE TITLE
Force mode: universal on 'nuxt generate'

### ISSUE THREAD

**Issue by hecktarzuli** [nuxt/nuxt.js#6945]
### What problem does this feature solve?

I've been trying to use nuxt for creating a static site, and it took me forever to figure out that I actually had to use mode:universal in order for it to actually work.

### What does the proposed changes look like?

OPTIONS

1) Throw an error when doing a 'nuxt generate' and the mode is 'spa' OR
2) auto change the runtime mode to 'universal' when 'nuxt generate' runs.

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10261">#c10261</a>)</em></sub></div>

**Comment by husayt** [comment]
This is most probably is not the issue with SPA mode in `nuxt generate`, but with your code. One thing needed to remember that each of the modes have their own specifics (which are often very confusing) which need to be taken care of when using with `nuxt generate`. For instance I have the project with opposite problem, where SPA works and Universal mode fails. 

There are pretty valid use cases for both modes  and they are both very critical. With all this said there are recognized issues and complexities around `nuxt generate` and this is going to be addressed. There is going to be a big addition/change to this space with [options.target feature](https://github.com/nuxt/nuxt.js/pull/6159). Now with long awaited new `fetch()` with server-side component rendering supported `options.target` seems will land soon. That will solve lots of problems you face and allow some new amazing functionality.

**Comment by hecktarzuli** [comment]
To my knowledge, SPA mode with nuxt generate, generates pages everywhere, but they are all exactly the same (and not really generated)

**Comment by husayt** [comment]
Yes, pretty much with adjustment for different entry points and that is the key thing which helps to imitate server side support for routing on static hosting. So you get SPA on static hosting with server side routing.

**Comment by hecktarzuli** [comment]
Right, so nuxt build = nuxt generate.. Hence why we'd want the feature requested.  But if all this will change with the target feature then this is all a moot point.

**Comment by husayt** [comment]
With `nuxt build` you still need `nuxt run` to run your website. With generate you get  static html/css/js ready to work with any http webserver (`http-serve`, netlify, assets of java webserver, ....). This is the key difference, `nuxt generate` output is completely nuxt agnostic, whereas you need `nuxt run` to serve `nuxt build` output.

**Comment by hecktarzuli** [comment]
Hmm, I don't think that's totally true.  `nuxt build` in spa mode doesn't need a server.

**Comment by husayt** [comment]
Here we come to the key moment in this discussion and I'm along the way also trying to clarify for myself what each of these modes do or supposed to do. 

You can have a nuxt built SPA with server. But if you don't need one, that means then most probably you just need to use `nuxt generate`. So opposite is probably true, `nuxt build` in spa mode without server doesn't make much sense as `nuxt run` will be serving static content, for which it is overkill.

The way I differentiate to myself  is this way: if I want to "eject" from nuxt and end up with generic html/css/js bundle then I need generate. If I still want nuxt to serve content then my answer is `nuxt build/run` pair.

**Comment by Atinux** [comment]
Actually `nuxt generate` will be renamed to `nuxt export` to make it more logicial that is intended to be used outside `nuxt start`.

You can export both a SPA or a SSG actually.

The issue actually it `nuxt build --spa` that creates the `dist/` directory (I wanted to mimic vue-cli behavior) but it makes no sense since some people use `nuxt start` even in spa (to keep serverMiddleware for instance).
